shared/components/ember-output.jsx

```diff
@@ -0,0 +1,90 @@
+/* global React */
+
+export default React.createClass({
+    displayName: 'EmberOutput',
+
+    propTypes: {
+        source  : React.PropTypes.string.isRequired,
+        context : React.PropTypes.object.isRequired
+    },
+
+    injectMessages: function (locale, messages) {
+        var container  = this.container;
+        var lookupName = locale.toLowerCase();
+
+        if (container.has('locale:' + lookupName)) {
+            container.unregister('locale:' + lookupName);
+        }
+
+        container.register('locale:' + lookupName, {
+            locale:   locale,
+            messages: messages
+        }, {
+            instantiate: false
+        });
```

Comment:
What's happening here with the `unregister` then `register`? Is this something an Ember dev would understand?